### PR TITLE
Fix element_identifier unavailable with data_collection input param

### PR DESCRIPTION
**Pull request by abretaud**
Here's a (failing) test case for galaxyproject/tools-iuc/issues/857 
I also added a small fix for an UnboundLocalError I encountered while writing it

### CODE REVIEW

**Comment by abretaud** [comment]
It should be fixed now, and it turns out it had nothing to do with conditionals

File: lib/galaxy/tools/wrappers.py [hunk 1]
```diff
@@ -307,7 +307,7 @@ def to_wrapper( dataset ):
                 element = dataset
                 dataset = element.dataset_instance
                 kwargs["identifier"] = element.element_identifier
-            return self._dataset_wrapper( dataset, dataset_paths, **kwargs )
+            return self._dataset_wrapper( dataset, dataset_paths, identifier = element_identifier, **kwargs )
```
Comment (nsoranzo) [inline lib/galaxy/tools/wrappers.py]: I don't think is needed, should be already done at line 309.

File: test/api/test_tools.py [hunk 2]
```diff
@@ -805,6 +806,39 @@ def test_identifier_with_multiple_normal_datasets( self ):
...
+            "input1": {'src': 'hdca', 'id': dataset_collection['id']},
+        }
+
+        self.dataset_populator.wait_for_history( history_id, assert_ok=True )
+        create_response = self._run( "identifier_conditional", history_id, inputs )
+        self._assert_status_code_is( create_response, 200 )
+        create = create_response.json()
+        outputs = create[ 'outputs' ]
+        jobs = create[ 'jobs' ]
+        implicit_collections = create[ 'implicit_collections' ]
```
Comment (nsoranzo) [inline test/api/test_tools.py]: This variable is never used, can be removed.

**Comment by mvdbeek** [comment]
This looks good to me, should we merge @nsoranzo ?

File: test/functional/tools/identifier_conditional.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<tool id="identifier_conditional" name="identifier_conditional">
```
Comment (nsoranzo) [inline test/functional/tools/identifier_conditional.xml]: Should this id be `identifier_collection`? Seems more proper and it's what is used at https://github.com/galaxyproject/galaxy/pull/2570/files#diff-46fef38dde830e533e91e849d89bceabR809

If yes, also this file should be renamed and the change propagated to `test/functional/tools/samples_tool_conf.xml`

File: test/functional/tools/identifier_conditional.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  <command>
+    #for $input in $input1:
+      echo '$input.element_identifier' >> 'output1';
+    #end for
+  </command>
+  <inputs>
+    <param type="data_collection" collection_type="list" name="input1" label="Input 1" multiple="true" />
```
Comment (nsoranzo) [inline test/functional/tools/identifier_conditional.xml]: Remove `multiple="true"` ? Or is this supporting multiple dataset lists?

**Comment by nsoranzo** [comment]
Thanks @abretaud!

**Comment by abretaud** [comment]
Thank you!

**Pull request merged**